action.yml

```
@@ -27,50 +27,60 @@ runs:
       # on FOSSA outages.
     - if: steps.analyze.outcome == 'success'
-      name: 'Run FOSSA Test'
+      name: 'Run `fossa test`'
       id: test
-      uses: fossas/fossa-action@5913e730490ebf75ae47b59687b7e590289eed92
-      with:
-        api-key: ${{ steps.set_key.outputs.key }}
-        run-tests: true
+      continue-on-error: true
+      env:
+        FOSSA_API_KEY: ${{ steps.set_key.outputs.key }}
+      shell: bash
+      run: |
+        exec &> >(tee -a "test_logs.txt")
+        # Set timeout to 5 minutes (default of 60 minutes is waaaay too long to block CI)
+        fossa test --timeout 300
 
-    - if: github.repository_owner == 'getsentry' && failure()
-      name: 'Handle errors'
+    - if: steps.analyze.outcome == 'failure' || steps.test.outcome == 'failure'
+      name: 'Send error to Sentry on `fossa-cli` errors'
       shell: bash
+      env:
+        SENTRY_DSN: https://decbca863c554db095624ede8a83310c@o1.ingest.sentry.io/4505031352713216
       run: |
-        echo
-        echo "🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 "
-        echo
-        echo "Eep! It seems that this PR introduces a license violation. Did you add any libraries? Do they use the GPL or some weird license? Am I a confused bot? If you need a hand, cc: @getsentry/open-source in a comment. 🙏"
-        echo
-        echo "🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 "
+        if [[ ${{ steps.analyze.outcome }} == 'failure' ]]; then
+          curl -sL https://sentry.io/get-cli/ | sh
+          # Environment variables will automatically be sent, so we just want some minimal information
+          error_msg=$(cat analyze_logs.txt | grep -zoP '(?<=>>> Relevant errors\n\n    Error\n\n      ).*?(?=\n)' || echo 'unknown error message')
+          sentry-cli send-event -m "FOSSA analyze error: $error_msg" -t repo:$GITHUB_REPOSITORY -e url:$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID --logfile analyze_logs.txt
+          exit 0
+        fi
+        if grep -q "The scan has revealed issues. Number of issues found:" test_logs.txt; then
+          echo
+          echo "🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 "
+          echo
+          echo "Eep! It seems that this PR introduces a license violation. Did you add any libraries? Do they use the GPL or some weird license? Am I a confused bot? If you need a hand, cc: @getsentry/open-source in a comment. 🙏"
+          echo
+          echo "🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 🛑 "
+          exit 1
+        fi
+        curl -sL https://sentry.io/get-cli/ | sh
+        error_msg=$(cat test_logs.txt | grep -zoP '(?<=>>> Relevant errors\n\n    Error\n\n      ).*?(?=\n)' || echo 'unknown error message')
+        sentry-cli send-event -m "FOSSA test error: $error_msg" -t repo:$GITHUB_REPOSITORY -e url:$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID --logfile test_logs.txt
```

Let's trim this up here and in the analyze case so we get more of the useful error message in the UI.

```suggestion
        sentry-cli send-event -m "test: $error_msg" -t repo:$GITHUB_REPOSITORY -e url:$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID --logfile test_logs.txt
```